assets/js/blocks/stock-filter/block.js

```diff
@@ -48,10 +34,31 @@ const StockStatusFilterBlock = ( {
 	attributes: blockAttributes,
 	isEditor = false,
 } ) => {
+	const hideOutOfStockItems = useMemo(
+		() => getSetting( 'hideOutOfStockItems', false ),
+		[]
+	);
+	const { outofstock, ...otherStockStatusOptions } = useMemo(
+		() => getSetting( 'stockStatusOptions', {} ),
+		[]
+	);
+	const STOCK_STATUS_OPTIONS = useMemo( () => {
+		return hideOutOfStockItems
+			? otherStockStatusOptions
+			: { outofstock, ...otherStockStatusOptions };
+	}, [ hideOutOfStockItems, otherStockStatusOptions, outofstock ] );
+
 	const [ checked, setChecked ] = useState( [] );
 	const [ displayedOptions, setDisplayedOptions ] = useState(
 		blockAttributes.isPreview ? previewOptions : []
 	);
+	// Filter added to handle if there are slugs without a corresponding name defined.
+	const initialOptions = useMemo( () => {
```

Comment:
Unless there are a lot of values within `STOCK_STATUS_OPTIONS` it doesn't seem worth using `useMemo`. Just going to quote Kent C Dodds:

```
Performance optimizations are not free. They ALWAYS come with a cost but do NOT always come with a benefit to offset that cost.

Therefore, optimize responsibly.
```
- [Source](https://kentcdodds.com/blog/usememo-and-usecallback)

Reply:
Under normal circumstances this will have a length of 3, but merchants _can_ add more stock statuses. After thinking about it, the complexity this brings to the component coupled with the overheads of `useMemo` this don't seem lik a worthwhile optimisation in my opinion.